src/libraries/Common/src/System/Security/Cryptography/X509Certificates/X509CertificateLoader.Pkcs12.cs

```diff
@@ -21,6 +22,9 @@ public static partial class X509CertificateLoader
         private const int NTE_FAIL = unchecked((int)0x80090020);
 #endif
 
+        [ThreadStatic]
+        private static HashSet<string>? t_attributeDuplicateCheck;
```

Comment:
It's not clear to me why this needs to be a `[ThreadStatic]`. It's used in exactly one place, `RejectDuplicateAttributes`, and it is `Clear`'ed every time. Why could we not just create a new `HashSet` every time `ProcessSafeContents` is called, and pass that in to `RejectDuplicateAttributes`?

Reply:
"Because we have to do it every time...".

But the question does imply "yeah, and how many PKCS#12 files does the average process open?"  Zero.  "If it's non-zero, how many does the average process open?"  One.  "How many were you thinking it opened when you wrote this?" Infinity, like the test suite.

Reply:
Ah, okay, it's a cache. That went over my head. For some reason I assumed there was some intentional preservation of the HashSet between calls or something, like we had with the work limiter.

If you want to leave it around as a cache, I suppose that is fine. A comment around that would be helpful.

I would think that it feels a little odd for us to be caching this in a `[ThreadStatic]`. There are currently [zero uses of thread static](https://github.com/search?q=repo%3Adotnet%2Fruntime+ThreadStatic+path%3Asrc%2Flibraries%2FSystem.Security.Cryptography*%2Fsrc%2F**&type=code) in S.S.Cryptography* and it feels like this is being specialized for no particular reason.